teuthology/task/ansible.py

```diff
@@ -0,0 +1,225 @@
+        """
+        repo = self.config.get('repo', '.')
+        if repo.startswith('http://') or repo.startswith('https://') or \
+                repo.startswith('git@'):
+            repo_path = fetch_repo(repo,
+                                   self.config.get('branch', 'master'),
+                                   )
+        else:
+            repo_path = os.path.abspath(os.path.expanduser(repo))
+        self.repo_path = repo_path
+
+    def get_playbook(self):
+        """
+        If necessary, fetch and read the playbook file
+        """
+        playbook = self.config['playbook']
+        if isinstance(playbook, list):
+            # Multiple plays in a list
+            self.playbook = playbook
+        elif isinstance(playbook, str) and (playbook.startswith('http://') or \
+                                            playbook.startswith('https://')):
+            response = requests.get(playbook)
+            response.raise_for_status()
+            self.playbook = yaml.safe_load(response.text)
+        elif isinstance(playbook, str):
+            try:
+                playbook_path = os.path.expanduser(playbook)
+                self.playbook_file = file(playbook_path)
+                playbook_yaml = yaml.safe_load(self.playbook_file)
+                self.playbook = playbook_yaml
+            except Exception:
+                log.error("Unable to read playbook file %s", playbook)
+                raise
+        else:
+            raise TypeError(
+                "playbook value must either be a list, URL or a filename")
+        log.info("Playbook: %s", self.playbook)
+
+    def get_inventory(self):
+        """
+        Determine whether or not we're using an existing inventory file
+        """
+        self.inventory = self.config.get('inventory')
+        etc_ansible_hosts = '/etc/ansible/hosts'
+        if self.inventory:
+            self.inventory = os.path.expanduser(self.inventory)
+        elif os.path.exists(etc_ansible_hosts):
+            self.inventory = etc_ansible_hosts
+        return self.inventory
+
+    def generate_hosts_file(self):
+        """
+        Generate a hosts (inventory) file to use. This should not be called if
+        we're using an existing file.
+        """
+        hosts = self.cluster.remotes.keys()
+        hostnames = [remote.name.split('@')[-1] for remote in hosts]
+        hostnames.sort()
+        hosts_str = '\n'.join(hostnames + [''])
+        hosts_file = NamedTemporaryFile(prefix="teuth_ansible_hosts_",
+                                        delete=False)
+        hosts_file.write(hosts_str)
+        hosts_file.flush()
+        self.generated_inventory = True
+        self.inventory = hosts_file.name
+
+    def generate_playbook(self):
+        """
+        Generate a playbook file to use. This should not be called if we're
+        using an existing file.
+        """
+        for play in self.playbook:
+            play['hosts'] = 'all'
```

Comment:
I understand why you do this, but a comment might help the next reader of this task.